zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/Interpreter.java

```diff
@@ -486,4 +494,70 @@ public static RegisteredInterpreter findRegisteredInterpreterByClassName(String
     }
     return null;
   }
+
+  public String getKerberosRefreshInterval() {
+    if (System.getenv("KERBEROS_REFRESH_INTERVAL") == null) {
+      return "1d";
+    } else {
+      return System.getenv("KERBEROS_REFRESH_INTERVAL");
+    }
+  }
+
+  public Integer kinitFailThreshold() {
+    if (System.getenv("KINIT_FAIL_THRESHOLD") == null) {
+      return 5;
+    } else {
+      return new Integer(System.getenv("KINIT_FAIL_THRESHOLD"));
+    }
+  }
+
+  public Long getTimeAsMs(String time) {
+    if (time == null) {
+      logger.error("Cannot convert to time value.", time);
+      time = "1d";
+    }
+
+    Matcher m = Pattern.compile("(-?[0-9]+)([a-z]+)?").matcher(time.toLowerCase());
+    if (!m.matches()) {
+      throw new IllegalArgumentException("Invalid time string: " + time);
+    }
+
+    long val = Long.parseLong(m.group(1));
+    String suffix = m.group(2);
+
+    if (suffix != null && !Constants.TIME_SUFFIXES.containsKey(suffix)) {
+      throw new IllegalArgumentException("Invalid suffix: \"" + suffix + "\"");
+    }
+
+    return TimeUnit.MILLISECONDS.convert(val,
+        suffix != null ? Constants.TIME_SUFFIXES.get(suffix) : TimeUnit.MILLISECONDS);
+  }
+
+  protected abstract boolean runKerberosLogin();
+
+  protected ScheduledExecutorService startKerberosLoginThread() {
```

Comment:
I kind of thought about where to put this, but then finally have `protected abstract boolean runKerberosLogin();`  and depending upon what all interpreters need to override this function will have this.

Like in this particular case (of shell interpreter) in the open method this function (startKerberosLoginThread) gets called, and `runKerberosLogin` function talks about how to kinit.

But sure I'm open for any other thoughts.